indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisorStateManager.java

```diff
@@ -0,0 +1,256 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.indexing.seekablestream.supervisor;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import org.apache.commons.lang.exception.ExceptionUtils;
+import org.apache.druid.indexer.TaskState;
+import org.apache.druid.indexing.seekablestream.common.StreamException;
+import org.apache.druid.java.util.common.DateTimes;
+import org.joda.time.DateTime;
+
+import java.util.ArrayList;
+import java.util.Deque;
+import java.util.List;
+import java.util.concurrent.ConcurrentLinkedDeque;
+
+public class SeekableStreamSupervisorStateManager
+{
+  public enum State
+  {
+    // Error states - ordered from high to low priority
```

Comment:
Yeah the priority is more implicit now - there is no overlap between `LOST_CONTACT_WITH_STREAM`, `UNABLE_TO_CONNECT_TO_STREAM`, and `UNHEALTHY_SUPERVISOR` - it reports one of the first two if the last exception thrown was wrapped in a `StreamException` (which happens for exceptions from calls made to the Kafka/Kinesis client library), otherwise it's `UNHEALTHY_SUPERVISOR`. Any of those 3 take priority over `UNHEALTHY_TASKS`.